test/BatchExchangeViewer.spec.ts

```diff
@@ -531,4 +532,39 @@ contract("BatchExchangeViewer [ @skip-on-coverage ]", (accounts) => {
       assert.equal(decodeIndexedOrders(result.elements).length, 1);
     });
   });
+  describe("getTokenInfo", () => {
+    it("Allows to get token address, symbol and decimals by ID", async () => {
+      const erc20detailed = await ERC20Detailed.at(token_1.address);
+      const symbolMethod = erc20detailed.contract.methods.symbol().encodeABI();
+      const decimalsMethod = erc20detailed.contract.methods
+        .decimals()
+        .encodeABI();
+
+      await token_1.givenMethodReturn(
+        symbolMethod,
+        web3.eth.abi.encodeParameter("string", "SCAM"),
+      );
+      await token_1.givenMethodReturnUint(decimalsMethod, 42);
+
+      const viewer = await BatchExchangeViewer.new(batchExchange.address);
+      const result = await viewer.getTokenInfo(1);
+      assert.equal(result[0], token_1.address);
+      assert.equal(result[1], "SCAM");
+      assert.equal(result[2].toNumber(), 42);
+    });
+    it("Reverts if token doesn't implement symbol or decimals", async () => {
```

Comment:
I don't really like that this method reverts when there is no symbol. This renders the method unusable in valid cases that can easily arise. Look at all the tokens without symbol on rinkeby for example. I think this would be more appropriate to simply return `undefined` considering that the only actual relevant information required here is token address and number of decimals. The symbol is simply aesthetics.

Reply:
Good point. Will wrap both calls in a solidity try/catch